File: Segregate 0s and 1s.java
class Solution {
    void segregate0and1(int[] arr) {
        
        int mid=0;
        int left=0;
        
        while(mid<arr.length){
            if(arr[mid]==0){
                int temp=arr[left];
                arr[left]=arr[mid];
                arr[mid]=temp;
                left++;
            }
            
            mid++;
        }
    }
}

File: Segregate 0s and 1s.cpp
class Solution {
  public:
    void segregate0and1(vector<int> &arr) {
        // code here
        int n=arr.size();
        int i=0, j=n-1;
        while(i<j){
            while(i<j && arr[i]==0){
                i++;
            }
            while(i<j && arr[j]==1){
                j--;
            }
            swap(arr[i], arr[j]);
            i++;
            j--;
        }
    }
};
